File: Practica1/Ej4/FilterManager.java
class FilterManager{

    private FilterChain filterchain;

    public FilterManager(ChatTarget target){
        filterchain=new FilterChain();
        filterchain.setTarget(target);
    }

    public void addFilter(Filter filter){
        filterchain.addFilter(filter);
    }

    public void haceCuenta(Cuenta cuenta){
        filterchain.comprueba(cuenta);
    }

}

File: Practica2/Ej1/lib/FilterManager.dart
import 'ChatTarget.dart';
import 'Filter.dart';
import 'FilterChain.dart';
import 'Cuenta.dart';

class FilterManager{
  FilterChain _filterChain = FilterChain();

  FilterManager(ChatTarget target){
    _filterChain.setTarget(target);
  }

  void addFilter(Filter filter){
    _filterChain.addFilter(filter);
  }

  String haceCuenta(Cuenta cuenta){
    return _filterChain.comprobar(cuenta);
  }

}